dataplug/app/com/hubofallthings/dataplug/dao/UserDAOImpl.scala

```diff
@@ -108,4 +109,27 @@ class UserDAOImpl @Inject() (protected val dbConfigProvider: DatabaseConfigProvi
 
     db.run(q).map(_ => Unit)
   }
+
+  /**
+   * Deletes user from relevant user tables.
+   *
+   * @param phata The phata of the user to delete.
+   * @param userId The userId of user to delete.
+   * @return Done if success.
+   */
+  def delete(phata: String, userId: String): Future[Done] = {
+    val dateNow = DateTime.now().toLocalDateTime.toString
+    val q = DBIO.seq(
+      Tables.DataplugUser.filter(_.phata === phata).delete, //phata
+      Tables.DataplugUserStatus.filter(_.phata === phata).delete, //phata
+      Tables.UserLink.filter(_.masterUserId === phata).delete, //phata
+      Tables.UserOauth2Info.filter(_.userId === userId).delete, //user id
+      Tables.UserUser.filter(_.userId === userId).delete, // user id
+      Tables.LogDataplugUser.filter(_.phata === phata).map(_.message).update(Some(s"Deleted on $dateNow")), // phata
```

Comment:
Here we log the deletion. I though to reuse the same table. Dunno if it's good or bad. Didn't find somewhere that this will cause issues. I may be wrong. If we think it's not sufficient then I will create a new table

Reply:
![Screenshot 2020-05-21 at 10 29 23](https://user-images.githubusercontent.com/11439452/82545130-19279400-9b4e-11ea-833c-7a368b200bec.png)
This what the table looks like